apps/heft/src/plugins/TypeScriptPlugin/EmitFilesPatch.ts

```diff
@@ -149,16 +133,22 @@ export class EmitFilesPatch {
           );
 
           emitSkipped = emitSkipped || flavorResult.emitSkipped;
+          for (const diagnostic of flavorResult.diagnostics) {
+            diagnostics.push(diagnostic);
+          }
```

Comment:
```suggestion
          diagnostics.push(...flavorResult.diagnostics);
```

Reply:
Runs risk of stack overflow if the number of diagnostics are large. Have had that problem occur before when trying to use this optimization, so I'd prefer to be safe.